cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -391,12 +495,20 @@ private WorkflowAction initialize(WorkflowActionSpecification actionSpec,
     WorkflowAction action = instantiator.get(TypeToken.of((Class<? extends WorkflowAction>) clz)).create();
 
     ClassLoader oldClassLoader = ClassLoaders.setContextClassLoader(action.getClass().getClassLoader());
+
+    BasicWorkflowContext workflowContext =
+      new BasicWorkflowContext(workflowSpec, actionSpec,
+                               logicalStartTime,
```

Comment:
`logicalStartTime` can go up in the same line as `workflowSpec` and `actionSpec`